apps/src/blockly/addons/cdoWorkspaceSvg.js

```diff
@@ -26,8 +50,23 @@ export default class WorkspaceSvg extends GoogleBlockly.WorkspaceSvg {
     /** @type {Blockly.Trashcan} */
     this.trashcan = new Blockly.Trashcan(this);
     var svgTrashcan = this.trashcan.createDom();
-    this.flyout_?.svgGroup_?.appendChild(svgTrashcan);
-    this.hideTrashcan();
+
+    switch (this.getToolboxType()) {
+      case ToolboxType.UNCATEGORIZED: {
+        const trashcanHolder = Blockly.utils.dom.createSvgElement('svg', {
+          id: 'trashcanHolder',
+          height: 125,
+          style: 'position: absolute; display: none;'
+        });
+        trashcanHolder.appendChild(svgTrashcan);
+        this.flyout_.svgGroup_.appendChild(trashcanHolder);
```

Comment:
i'm not sure of a better way to do this, but this line tightly couples `ToolboxType.UNCATEGORIZED` to the presence of `this.flyout_`. maybe we abstract this functionality (lines 56-62) into a function next to `getToolboxType` and add a comment about this coupling? i say that because i know we're on a tight deadline 😄 but we can discuss how to handle this long term

Reply:
i think long term, there would be something like an `appendTrashcanHolder` function that lives on the Flyout class, but i think that class is in the blockly repo? then here we could do something like:

```javascript
// cdoWorkspaceSvg.js
getToolbox() {
 return this.flyout_ || this.toolbox_;
}

addTrashcan() {
  this.getToolbox().appendTrashcanHolder?.(svgTrashcan);
}
```

and the Flyout class would have that method, implementing the functionality here, whereas the Toolbox class would no-op or just not have that method. we also might want to find an elegant way to share the creation of the trashcanHolder (via `Blockly.utils.dom.createSvgElement`) since that's the same in either case

Reply:
and i think we could do the same sort of refactor for the functionality that calculates the toolboxWidth in cdoTrashcan